compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -247,13 +248,11 @@ data ModuleContents = ModuleContents
   , mcTypeDefs :: [TyThing]
   , mcTemplateBinds :: MS.Map TypeConName TemplateBinds
   , mcExceptionBinds :: MS.Map TypeConName ExceptionBinds
+  , mcInterfaceBinds :: MS.Map TypeConName InterfaceBinds
+    -- ^ Maps an interface to the contents of its definition.
+  , mcInterfaceInstanceBinds :: MS.Map TypeConName InterfaceInstanceGroup
+    -- ^ Maps a template to the interface instances defined in it.
   , mcChoiceData :: MS.Map TypeConName [ChoiceData]
-  , mcImplements :: MS.Map TypeConName [(Maybe LF.SourceLoc, GHC.TyCon)]
-  , mcRequires :: MS.Map TypeConName [(Maybe LF.SourceLoc, GHC.TyCon)]
-  , mcInterfaceMethodInstances :: MS.Map (GHC.Module, TypeConName, TypeConName) [(T.Text, GHC.Expr GHC.CoreBndr)]
-  , mcInterfaces :: MS.Map TypeConName GHC.TyCon
-  , mcInterfaceViews :: MS.Map TypeConName GHC.Type
-  , mcInterfaceViewInstances :: MS.Map (GHC.Module, TypeConName, TypeConName) [GHC.Expr GHC.CoreBndr]
```

Comment:
This is the core change:

* `mcInterfaceBinds` replaces:
  * `mcInterfaces`
  * `mcRequires`
  * `mcInterfaceView`
  * and a direct use of `mcBinds` to extract the interface methods

* `mcInterfaceInstanceBinds` replaces:
  * `mcImplements`
  * `mcInterfaceMethodInstances`
  * `mcInterfaceViewInstances`

Additionally, `mcInterfaceInstanceBinds` is structured differently compared with the fields it replaces:
* (old) `mcImplements`: map from templates to implemented interfaces
* (old) `mcInterfaceMethodInstances`: map from (interface, template) tuple to method implementations
* (old) `mcInterfaceViewInstances`: as above but mapping to view implementations 
* (new) `mcInterfaceInstanceBinds`: map from templates to interface instances inside them
* (future) `mcInterfaceInstanceBinds`: map from templates _or interfaces_ to interface instances inside them